chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -111,67 +112,71 @@ func New(node *neutrino.ChainService, spendHintCache chainntnfs.SpendHintCache,
 // Start contacts the running neutrino light client and kicks off an initial
 // empty rescan.
 func (n *NeutrinoNotifier) Start() error {
-	// Already started?
-	if atomic.AddInt32(&n.started, 1) != 1 {
-		return nil
-	}
+	var startErr error
+
+	n.start.Do(func() {
+		// Start our concurrent queues before starting the rescan, to ensure
+		// onFilteredBlockConnected and onRelavantTx callbacks won't be
+		// blocked.
+		n.chainUpdates.Start()
```

Comment:
We could extract all of this into an unexported method to avoid the additional indentation.